fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultScheduledDateGenerator.java

```diff
@@ -104,6 +104,12 @@ public LocalDate generateNextRepaymentDate(final LocalDate lastRepaymentDate, fi
             }
         }
 
+        final LocalDate maxDateForFixedLength = loanApplicationTerms.calculateMaxDateForFixedLength();
+        // Fixed Length validation
+        if (maxDateForFixedLength != null && DateUtils.isAfter(dueRepaymentPeriodDate, maxDateForFixedLength)) {
```

Comment:
I dont think this is righ. I reckon we might have validation to avoid such situation where we are adjusting not the last repayment, but still this is not checking whether it is the latest or not. 

Also i dont think we need this here... Its more than enough to adjust the loan end date and the very last installment due date.

Reply:
Done! updated

Reply:
I dont see any changes here... why do we need this for the "generateNextRepaymentDate"?

I dont think we need this at all. Please let me know if i am mistaken.